MiniBootcamp/MiniBootcamp/FeedDataManager.swift

```diff
@@ -8,11 +8,97 @@
 import Foundation
 
 protocol FeedDataManagerProtocol {
-    func fetch(completion: (Result<[TweetCellViewModel], Error>) -> Void)
+    func fetch(completion: @escaping (Result<[TweetCellViewModel], Error>) -> Void)
+}
+
+enum TweetAPIError: Error {
+    case noData
+    case response
+    case parsingData
 }
 
 class FeedDataManager: FeedDataManagerProtocol {
-    func fetch(completion: (Result<[TweetCellViewModel], Error>) -> Void) {
-        completion(.failure(NSError(domain: "", code: 0)))
+  let session: URLSession
+
+  init(session: URLSession = URLSession.shared) {
+    self.session = session
+  }
+
+    func fetch(completion: @escaping (Result<[TweetCellViewModel], Error>) -> Void) {
+      let url = URL(string: "https://gist.githubusercontent.com/ferdelarosa-wz/0c73ab5311c845fb7dfac4b62ab6c652/raw/6a39cffe68d87f1613f222372c62bd4e89ad06fa/tweets.json")
+      guard let url else { return }
+
+      let request = URLRequest(url: url)
```

Comment:
the request should be a parameter for the fetch method. I understand that it is just one `url` for this project but I just trying to do this in a general way.